### PR TITLE
Small library update plus general bugfixes

### PR DESCRIPTION
**Pull request by jfleming9357**
- Fixed a bug where authors was not always an array in BookDetail.jsx and could crash the site
- Removed excessively large chunks of commented code
- Fixed a react warning where some clubs would have the same key value in userBookClubs.jsx
- Modified the /user/book endpoint to use a get request correctly (removed usage of req.body and changed to req.params)
- port is now back to 3000 (my b)
- Library now shows owned books by default, and marks them as owned in search results
- Changed recommended book carousel to gallery
-recommended now uses a random genre from the bestseller list

### CODE REVIEW

File: client/components/global/BookDetail.jsx [hunk 1]
```diff
@@ -60,7 +28,7 @@ export default function BookDetail({
             <Image src={book.imageURL} fluid rounded />
           </Col>
           <Col>
-            <div>by {book.authors.join(', ')}</div>
+            <div>by {Array.isArray(book.authors) ? book.authors.join(', ') : book.authors}</div>
```
Comment (daniel-he-dev) [inline client/components/global/BookDetail.jsx:31]: google always returns an array. should not be a problem anymore once my change gets merged. significantly changes bookdetail

File: server/server.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ require('dotenv').config();
 const app = express();
 const dbURI = `mongodb+srv://jfleming9357:${process.env.MONGO_PASS}@cluster0.v4rli.mongodb.net/papercut?retryWrites=true&w=majority`;
 
-const port = process.env.PORT || 3008;
+const port = process.env.PORT || 3000;
```
Comment (daniel-he-dev) [inline server/server.js:15]: flip flopping so much

File: server/server.js [hunk 3]
```diff
@@ -66,7 +66,6 @@ app.get('/checkauth', isAuthenticated, function (req, res) {
   Subscription.findOne({ _id: req.user.subscriptionTier })
     .then((subscription) => {
       req.user._doc.subscription = subscription;
-      console.log(req.user);
```
Comment (daniel-he-dev) [inline server/server.js:69]: love it

**Review by daniel-he-dev — COMMENTED** [review]
Merge my PR first, merge in changes. Then should be good

**Comment by ShinanSun** [comment]
@jfleming9357 you might need to pull from main again since Daniel just did a merge and has changed the bookDetails model.

File: client/components/profilePage/myLibrary.jsx [hunk 4]
```diff
@@ -1,75 +1,63 @@
...
-      ratingCount: 7,
-      datePurchased: '02/15/2021'
-    }
-  ];
-
-  const [booksOwned, setBooksOwned] = useState(books);
+export default function myLibrary({ user }) {
+  const [booksOwned, setBooksOwned] = useState([]);
   const [searchInput, setSearchInput] = useState('');
   const [clickedBook, setClickedBook] = useState();
```
Comment (ShinanSun) [inline client/components/profilePage/myLibrary.jsx:12]: Don't know if there is gonna be a bug is you don't set initial value of clickedBook, maybe more safe to do `userState({})`

File: server/server.js (hunk 2 above)
Comment (jfleming9357) [reply server/server.js:15]: I keep trying to set it to 3000 but it gets changed back. I'm just trying to fix the problem I created :(

File: client/components/profilePage/myLibrary.jsx (hunk 4 above)
Comment (jfleming9357) [reply client/components/profilePage/myLibrary.jsx:12]: Debugging some interactions with the new book details and that is actually a change I made. Good catch.

**Review by ShinanSun — CHANGES_REQUESTED** [review]
@jfleming9357 you might need to pull from main again since Daniel just did a merge and has changed the bookDetails model. I saw you are using Dan's bookDetail component, maybe double check if there any changes need to be made after mergred his changes.

**Comment by jfleming9357** [comment]
> @jfleming9357 you might need to pull from main again since Daniel just did a merge and has changed the bookDetails model. I saw you are using Dan's bookDetail component, maybe double check if there any changes need to be made after mergred his changes.

Already have, but a LOT of bugs have popped up so it might be a bit before I merge

**Pull request merged**